hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/InodeTree.java

```diff
@@ -284,7 +290,15 @@ boolean isInternalDir() {
       return false;
     }
 
-    public T getTargetFileSystem() {
+    public T getTargetFileSystem() throws IOException {
```

Comment:
This is now doing more than get. Needs javadocs. What is the thread safety story here?

Reply:
Added javadoc. Also added synchronized block around apply call to make sure only one thread is initializing the file system instance.